src/Nethermind/Nethermind.Evm.Test/Eip6780Tests.cs

```diff
@@ -19,133 +19,145 @@
-                        .PushData(1)
-                        .Op(Instruction.SSTORE)
-                        .PushData(40)
-                        .Op(Instruction.JUMP)
-                        .Op(Instruction.JUMPDEST)
-                        .PushData(TestItem.PrivateKeyB.Address)
-                        .Op(Instruction.SELFDESTRUCT)
-                        .Op(Instruction.JUMPDEST)
-                        .Done;
-            byte[] initCode = Prepare.EvmCode.ForInitOf(contractCode).Done;
             byte[] salt = new UInt256(123).ToBigEndian();
             Address createTxAddress = ContractAddress.From(TestItem.PrivateKeyA.Address, 0);
-            Address contractAddress = ContractAddress.From(createTxAddress, salt, initCode);
+            _contractAddress = ContractAddress.From(createTxAddress, salt, _selfDestructCode);
             byte[] tx1 = Prepare.EvmCode
-                .Create2(initCode, salt, 99.Ether())
-                .Call(contractAddress, 100000)
-                .Call(contractAddress, 100000)
+                .Create2(_selfDestructCode, salt, 99.Ether())
                 .STOP()
                 .Done;
 
-            long gasLimit = 1000000;
+            Transaction createTx = Build.A.Transaction.WithCode(tx1).WithValue(100.Ether()).WithGasLimit(_gasLimit).SignedAndResolved(_ecdsa, TestItem.PrivateKeyA).TestObject;
+            Block block = Build.A.Block.WithNumber(BlockNumber)
+                .WithTimestamp(Timestamp)
+                .WithTransactions(createTx).WithGasLimit(2 * _gasLimit).TestObject;
 
-            EthereumEcdsa ecdsa = new(1, LimboLogs.Instance);
-            Transaction createTx = Build.A.Transaction.WithCode(tx1).WithValue(100.Ether()).WithGasLimit(gasLimit).SignedAndResolved(ecdsa, TestItem.PrivateKeyA).TestObject;
-            Block block = Build.A.Block.WithNumber(blockNumber)
-                .WithTimestamp(timestamp)
-                .WithTransactions(createTx).WithGasLimit(2 * gasLimit).TestObject;
+            _processor.Execute(createTx, block.Header, NullTxTracer.Instance);
+
+            AssertDestroyed();
+            AssertSendAll();
+        }
+
+        [Test]
+        public void self_destruct_in_initcode_of_create_tx()
+        {
+            Transaction createTx = Build.A.Transaction.WithCode(_selfDestructCode).WithValue(99.Ether()).WithGasLimit(_gasLimit).SignedAndResolved(_ecdsa, TestItem.PrivateKeyA).TestObject;
+            Block block = Build.A.Block.WithNumber(BlockNumber)
+                .WithTimestamp(Timestamp)
+                .WithTransactions(createTx).WithGasLimit(2 * _gasLimit).TestObject;
 
             _processor.Execute(createTx, block.Header, NullTxTracer.Instance);
 
-            AssertStorage(new StorageCell(contractAddress, 1), 0);
+            AssertDestroyed();
+            AssertSendAll();
+        }
+
+        private void AssertNotDestroyed()
+        {
+            AssertCodeHash(_contractAddress, Keccak.Compute(_selfDestructCode.AsSpan()));
```

Comment:
why doesn't this do the opposite of ``AssertDestroyed`` ?

Reply:
Because with this call, i can make sure the contract has the keccak of the deployed code. but I am not sure how useful that is.